### PR TITLE
some extensions

### PR DESCRIPTION
**Pull request by soef**
discovery also works on windows

### CODE REVIEW

**Comment by PSeitz** [comment]
Thanks, that works.
What is the waitForNotify function for?

**Comment by PSeitz** [comment]
Do you need the reFriendlyName info, or is this just for debugging?

File: yamaha.js [hunk 1]
```diff
@@ -9,18 +9,68 @@ var chainedCommands = require('./chainedCommands');
...
+        callback(ip, info);
+    }
+    
+    peer.on("ready", function () {
+        peer.search({ ST: 'urn:schemas-upnp-org:device:MediaRenderer:1' });
+    }).on("found", function (headers, address) {
+        if (headers.LOCATION) {
+            request(headers.LOCATION, function (error, response, body) {
+                if (!error && response.statusCode == 200 && reYamahaManufacturer.test(body)) {
+                    var info = reFriendlyName.exec(body);
```
Comment (PSeitz) [inline yamaha.js:49]: Do you need the reFriendlyName info, or is this just for debugging?

File: simpleCommands.js [hunk 2]
```diff
@@ -128,6 +128,61 @@ Yamaha.prototype.setInputTo = function(to, zone){
...
+};
+
+Yamaha.prototype.setPureDirect = function(bo){
+    return this.sendCommand('<Sound_Video><Pure_Direct><Mode>' + (bo ? 'On' : 'Off') + '</Mode></Pure_Direct></Sound_Video>');
+};
+Yamaha.prototype.setHDMIOutput = function(no, bo){
+    return this.sendCommand('<Sound_Video><HDMI><Output><OUT_' + no + '>' + (bo ? 'On' : 'Off') + '</OUT_' + no + '></Output></HDMI></Sound_Video>', 'System');
+};
+Yamaha.prototype.scene = function(no) {
+    adapter.setState('scene', '', true);
```
Comment (PSeitz) [inline simpleCommands.js:164]: apadater is not defined

File: yamaha.js (hunk 1 above)
Comment (soef) [reply yamaha.js:49]: I don't need the friendly Name, bud it is nice to have. My be there are more than one device.

File: simpleCommands.js (hunk 2 above)
Comment (soef) [reply simpleCommands.js:164]: Ups, you can delete that line.

**Comment by soef** [comment]
The waitForNotify function is to be notified, if the yamaha device was removed from power. If you call that function the callback is called when a notify messae ist send by the yamaha receiver 10-20 seconds after it gets back power. So you don't neet to try to reconnect when the yamaha device is powerless.

**Comment by soef** [comment]
Can you please take care of the possibility to not catch request exceptions? My way was to use dontCatchRequestErrors. Otherwise the application will get a different exception because of a call to toString from a null object.

File: yamaha.js (hunk 1 above)
Comment (PSeitz) [reply yamaha.js:49]: I would rather add this as optional additional  filter for the query.

**Comment by PSeitz** [comment]
I added a catchRequestErrors flag, which is on true by default (97d7345)
The waitForNotify seems more like a general purpose query and not something related to the yamaha discovery/api

**Pull request merged**